File: Test_HighPass.py
import rtAttenPy
import rtAttenPy.highpass
import scipy.io
import numpy as np
from contexttimer import Timer

# Get highpass.tgz and tar zxvf
a = scipy.io.loadmat('highpass_in.mat')
b = scipy.io.loadmat('highpass_out.mat')

data = a['a']['data'][0][0]

with Timer() as t:
    result = rtAttenPy.highpass.highpass(np.transpose(data), 28)
print('New: %0.2fs' % t.elapsed)
np.testing.assert_array_almost_equal(np.transpose(result), b['b'])

File: matlab/Test_HighPass.m
load highpass_in.mat
load highpass_out.mat

tic
result = HighPassBetweenRuns(a.data, a.TR, a.cutoff);
toc

all(all(result == b))

result = HighPassRealTime(a.data, a.TR, a.cutoff);